backend/api/visualization.proto

```diff
@@ -0,0 +1,120 @@
+
+syntax = "proto3";
+
+option go_package = "github.com/kubeflow/pipelines/backend/api/go_client";
+package api;
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+import "backend/api/parameter.proto";
+import "backend/api/error.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  responses: {
+    key: "default";
+    value: {
+      schema: {
+        json_schema: {
+          ref: ".api.Status";
+        }
+      }
+    }
+  }
+  // Use bearer token for authorizing access to job service.
+  // Kubernetes client library(https://kubernetes.io/docs/reference/using-api/client-libraries/)
+  // uses bearer token as default for authorization. The section below
+  // ensures security definition object is generated in the swagger definition.
+  // For more details see https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#securityDefinitionsObject
+  security_definitions: {
+    security: {
+      key: "Bearer";
+      value: {
+        type: TYPE_API_KEY;
+        in: IN_HEADER;
+        name: "authorization";
+      }
+    }
+  }
+  security: {
+    security_requirement: {
+      key: "Bearer";
+      value: {};
+    }
+  }
+};
+
+service VisualizationService {
+  rpc GenerateVisualization(GenerateVisualizationRequest) returns (Visualization) {
+    option (google.api.http) = {
+      post: "/apis/v1beta1/visualizations"
+      body: "visualizaiton"
+    };
+  }
+}
+
+// Create visualization by providing the pipeline ID, run ID, component ID,
+// type of visualization that is desired, and input data paths. Should a custom
+// visualization be specified, the customVisualization field (an array of python
+// commands) is also required.
+message GenerateVisualizationRequest {
+  Visualization visualization = 1;
+}
+
+message Visualization {
+  // Output. Unique visualization ID. Generated by API server.
+  string id = 1;
+
+  // ID of pipeline that visualization is associated with.
+  // This is required when creating the pipeline through GenerateVisualization
+  // API.
+  string pipeline = 2;
+
+  // ID of the run that visualization is associated with.
+  // This is required when creating the pipeline through GenerateVisualization
+  // API.
+  string run = 3;
+
+  // ID of the component that visualization is associated with.
+  // This is required when creating the pipeline through GenerateVisualization
+  // API.
+  string component = 4;
+
+  // Type of visualization to be generated.
+  // This is required when creating the pipeline through GenerateVisualization
+  // API.
+  enum Type {
+    CUSTOM = 0;
+    ROC_CURVE = 1;
+  };
+  Type type = 5;
+
+  // Array of paths of data to be used during generation of visualizations.
+  // This is required when creating the pipeline through GenerateVisualization
+  // API.
+  repeated string inputPaths = 6;
```

Comment:
does it need to be an array?

Reply:
It could be a comma separated string. Allowing it to be an array just provides support for multiple inputs.

Reply:
do we have an example of multiple inputs?

Reply:
The current [roc curve visualization](https://github.com/kubeflow/pipelines/blob/master/components/local/roc/src/roc.py#L65) appears to support multiple files.